src/content-tags-drawer/ContentTagsCollapsible.jsx

```diff
@@ -216,8 +215,8 @@ ContentTagsCollapsible.propTypes = {
       value: PropTypes.string,
       lineage: PropTypes.arrayOf(PropTypes.string),
     })),
+    canTagObject: PropTypes.bool.isRequired,
   }).isRequired,
-  editable: PropTypes.bool.isRequired,
 };
```

Comment:
Some of these type definition are repeated in multiple places, it might be worth defining them once and them reusing them. 

i.e. you could do put into a `data/proptypes.js` the following:

```js

import PropTypes from 'prop-types';

const tag = PropTypes.shape({
  value: PropTypes.string,
  lineage: PropTypes.arrayOf(PropTypes.string),
});

const taxonomy = PropTypes.shape({
  id: PropTypes.number,
  name: PropTypes.string,
  contentTags: PropTypes.arrayOf(tag),
  canTagObject: PropTypes.bool.isRequired,
});

export default {
  tag,
  taxonomy,
};
```

Then they can be used anywhere as:

```js
import TaxonomyProps from './data/proptypes';

ContentTagsCollapsible.propTypes = {
  contentId: PropTypes.string.isRequired,
  taxonomyAndTagsData: TaxonomyProps.taxonomy.isRequired,
};
```

This will avoid needing to update all places when the structure changes, and also will also make it easier to discover all the places that rely on this data structure.

Reply:
Thank you for your review @xitij2000 ! And yep, we have to wait for https://github.com/openedx/edx-platform/pull/34004 to merge before merging here.

> Some of these type definition are repeated in multiple places, it might be worth defining them once and them reusing them.

Unfortunately, it looks like the components in src/taxonomy and src/content-tags-drawer don't share enough propTypes to warrant creating a `prototypes.js` file for them to use.

However, I do think it's silly that our test modules don't just re-use the propTypes of the components they're testing, so I've fixed that: [172add8](https://github.com/openedx/frontend-app-course-authoring/pull/787/commits/172add88e5b1ceb0bea549e9ddeaf09afdaedbcd)

Let me know what you think?

Reply:
Sure, I think that makes sense. I don't see any benefit in duplicating this